lib/plugins/aws/invoke/index.js

```diff
@@ -24,43 +24,44 @@ class AwsInvoke {
   }
 
   extendedValidate() {
-    this.validate();
-
-    // validate function exists in service
-    this.options.functionObj = this.serverless.service.getFunction(this.options.function);
-    this.options.data = this.options.data || '';
-
-    return new BbPromise(resolve => {
-      if (this.options.data) {
-        resolve();
-      } else if (this.options.path) {
-        const absolutePath = path.isAbsolute(this.options.path) ?
-          this.options.path :
-          path.join(this.serverless.config.servicePath, this.options.path);
-        if (!this.serverless.utils.fileExistsSync(absolutePath)) {
-          throw new this.serverless.classes.Error('The file you provided does not exist.');
+    return this.validate().then(() => {
+      try {
+        // validate function exists in service
+        this.options.functionObj = this.serverless.service.getFunction(this.options.function);
+        this.options.data = this.options.data || '';
+      } catch (error) {
+        return BbPromise.reject(error);
+      }
+
+      return new BbPromise(resolve => {
```

Comment:
Same here, there seems to be no need for this promise construction

Reply:
Yes, I agree. This is basically the old code and I didn't want to tweak it that much because I'm not certain whether our tests are fully functional.